scopes/lanes/ui/hooks/lanes-query/lanes-query.ts

```diff
@@ -0,0 +1,35 @@
+import { useDataQuery } from '@teambit/ui-foundation.ui.hooks.use-data-query';
+import { gql, QueryResult } from '@apollo/client';
+import { LanesModel, LanesQueryResult } from '@teambit/lanes.ui.lanes';
+import { useScopeQuery } from '@teambit/scope.ui.hooks.use-scope';
+
+const GET_LANES = gql`
+  {
+    lanes {
+      getLanes {
+        name
+        remote
+        isMerged
+        components {
+          id {
+            name
+            scope
+          }
+          head
+        }
+      }
+      getCurrentLaneName
+    }
+  }
+`;
+
+export function useLanes(): { lanesModel: LanesModel } & QueryResult<LanesQueryResult> {
+  const { data, ...rest } = useDataQuery<LanesQueryResult>(GET_LANES);
+  const { scope, loading } = useScopeQuery();
```

Comment:
it doesnt make sense in a workspace
this is only when on a scope
cant use context because this runs in the provider of the lanes context before the scope context has been hydrated, so have no choice but to query